gazebo_ros/launch/gazebo_launch.py

```diff
@@ -0,0 +1,99 @@
+Launch Gazebo with an empty world and initialize ROS with command line arguments.
+
+Search for model, plugin and media paths exported by packages and add them to
+Gazebo environment variables.
+
+e.g.  <export>
+          <gazebo_ros gazebo_model_path="${prefix}/models"/>
+          <gazebo_ros gazebo_media_path="${prefix}/models"/>
+      </export>
+
+${prefix} is replaced by package's share directory in install.
+Thus the required directory needs to be installed from CMakeLists.txt
+
+e.g.  install(DIRECTORY models
+          DESTINATION share/${PROJECT_NAME})
+"""
+
+import os
+
+from ament_index_python.packages import get_package_share_directory
+from catkin_pkg.package import InvalidPackage, PACKAGE_MANIFEST_FILENAME, parse_package
+import launch
+from ros2pkg.api import get_package_names
+
+
+def gazebo_ros_paths():
+    gazebo_model_path = []
+    gazebo_plugin_path = []
+    gazebo_media_path = []
+
+    for package_name in get_package_names():
+        package_share_path = get_package_share_directory(package_name)
+        package_file_path = os.path.join(package_share_path, PACKAGE_MANIFEST_FILENAME)
+        if os.path.isfile(package_file_path):
+            try:
+                package = parse_package(package_file_path)
+            except InvalidPackage:
+                continue
+            for export in package.exports:
+                if export.tagname == 'gazebo_ros':
+                    if 'gazebo_model_path' in export.attributes:
+                        plugin_xml_path = export.attributes['gazebo_model_path']
+                        plugin_xml_path = plugin_xml_path.replace('${prefix}', package_share_path)
+                        gazebo_model_path.append(plugin_xml_path)
+                    if 'plugin_path' in export.attributes:
+                        plugin_xml_path = export.attributes['plugin_path']
+                        plugin_xml_path = plugin_xml_path.replace('${prefix}', package_share_path)
+                        gazebo_plugin_path.append(plugin_xml_path)
+                    if 'gazebo_media_path' in export.attributes:
+                        plugin_xml_path = export.attributes['gazebo_media_path']
+                        plugin_xml_path = plugin_xml_path.replace('${prefix}', package_share_path)
+                        gazebo_media_path.append(plugin_xml_path)
+
+    gazebo_model_path = ':'.join(gazebo_model_path)
+    gazebo_plugin_path = ':'.join(gazebo_plugin_path)
+    gazebo_media_path = ':'.join(gazebo_media_path)
+
+    return gazebo_model_path, gazebo_plugin_path, gazebo_media_path
+
+
+def generate_launch_description():
+    os.system('. /usr/share/gazebo/setup.sh')
+
+    model, plugin, media = gazebo_ros_paths()
+    model += ':'+os.environ['GAZEBO_MODEL_PATH']
+    plugin += ':'+os.environ['GAZEBO_PLUGIN_PATH']
+    media += ':'+os.environ['GAZEBO_RESOURCE_PATH']
+
+    world_argument = launch.actions.DeclareLaunchArgument(
+        'world',
+        default_value='worlds/empty.world')
```

Comment:
The default value could be an empty string. This way we're sure we're using Gazebo's default value, in case that ever changes.